src/julia/Prechelt/src/Prechelt.jl

```diff
@@ -0,0 +1,76 @@
+#=
+# Port of Peter Norvig's Common Lisp program from http://norvig.com/java-lisp.html.
+#
+# - Julia version: 1.6.2
+# - Author: Renato Athaydes
+# - Date: 2021-07-24
+=#
+module Prechelt
+
+export run_encodings
+
+const emptyStrings = String[]
+
+const KEYS = begin
+    m = [
+         'j' 'r' 'd' 'f' 'a' 'c' 'b' 'l' 'g'
+         'n' 'w' 's' 't' 'm' 'i' 'k' 'o' 'h'
+         'q' 'x' 'y' '_' '_' 'v' 'u' 'p' 'z'
+        ]
+    map('a':'z') do c
+        n = 1
+        for col in eachcol(m) if c in col return n else n += 1 end end
+        0
+    end
+end
+
+function printTranslations(dict, num, digits, start=1, words=String[])
+    if start > length(digits)
+       return println(num, ": ", join(words, " "))
+    end
+    foundWord = false
+    n = BigInt(1)
+    for i in start:length(digits)
+        n = n * 10 + nth_digit(digits, i)
+        for word in get(dict, n, emptyStrings)
+            foundWord = true
+            printTranslations(dict, num, digits, i + 1, [words; word])
```

Comment:
You missed the fact that `[words; word]`, despite looking like the original Lisp `(cons words word)`, is extremely inefficient as it allocates a new array every time?!
Other languages are doing this instead (which unfortunately requires a mutable list):

```rust
words.push(word);
printTranslations(...);
words.pop();
```